src/schemathesis/specs/openapi/_adapter.py

```diff
@@ -25,9 +27,54 @@ def prepare_parameters(item: PathItem | Operation, *, resolver: RefResolver, bun
             # Copy the definition and bundle the schema to make it self-contained
             definition = {k: v for k, v in definition.items() if k != "schema"}
             try:
-                definition["schema"] = bundler.bundle(schema, resolver)
+                definition["schema"] = bundler.bundle(schema, resolver, inline_recursive=True)
             except BundleError as exc:
                 location = parameter.get("in", "")
                 name = parameter.get("name", "<UNKNOWN>")
                 raise InvalidSchema.from_bundle_error(exc, location, name) from exc
         yield definition
+
+
+@dataclass
+class Response:
+    """OpenAPI response definition."""
+
+    status_code: str
+    definition: dict[str, Any]
+    # TODO: Use types from `core.jsonschema`
+    schema: dict[str, Any] | bool | None
+
+    __slots__ = ("status_code", "definition", "schema")
+
+
+@dataclass
+class Responses:
+    """Collection of OpenAPI response definitions."""
+
+    _inner: dict[str, Response]
+
+    __slots__ = ("_inner",)
+
+    def find_by_status_code(self, status_code: int) -> Response | None:
+        """Find the most specific response definition matching the given HTTP status code."""
+        return _find_by_status_code(self._inner, status_code)
+
+
+T = TypeVar("T")
```

Comment:
It may be unnecessary to have this generic in the future - the function below is needed for inference and works with `dict`, but it actually can work with `Response` and inject links there. Add a note for the future